src/Juvix/Core/MainLang.hs

```diff
@@ -87,25 +108,45 @@ data Value
   | VNeutral Neutral
   | VNat Natural
 
-showVal ∷ Value → String
-showVal (VLam f) = showFun f
-showVal (VStar i) = "* " ++ show i
-showVal VNats = "Nats "
-showVal (VPi n v f) = "[" ++ show n ++ "] " ++ showVal v ++ " -> " ++ showFun f
-showVal (VPm n v f) =
-  "([" ++ show n ++ "] " ++ showVal v ++ ", " ++ showFun f ++ ") "
-showVal (VPa _ _ _) = "/\\ "
-showVal (VNPm _ _) = "\\/ "
-showVal (VNeutral _n) = "neutral "
-showVal (VNat i) = show i
-
-showFun ∷ (Value → Value) → String
-showFun _f = "\\x.t "
+varX ∷ Value
+varX = VNeutral (NFree (Global "x"))
+
+--(Value -> Value) are equal when given an arbitrary value they return the same value.
+valueToValueEq ∷ (Value → Value) → (Value → Value) → Bool
```

Comment:
Hmm I don't think this is true in general; what if they only returned the same value for `varX`?

Reply:
I think you might want to use `quote0` and check equality of the resulting `CTerm`.

Reply:
I can use ```quote0``` to check the equality of the resulting ```CTerm``` like this:

 ```valueToValueEq y1 y2 = quote0 (y1 varX) == quote0 (y2 varX)```

But I still don't have a solution for the concern that they may only return the same value for ```varX```. Any idea? Checking all possible ```Value``` type will result in infinite loops for the types with ```Value -> Value```.

Reply:
I mean use `quote0` first on the top-level value, e.g.:

```haskell
instance Eq Value where
  x == y = quote0 x == quote0 y
```

Would that work?

Reply:
I see!  Yes! Thank you!